compiler/rustc_codegen_llvm/src/debuginfo/gdb.rs

```diff
@@ -96,31 +100,14 @@ pub(crate) fn needs_gdb_debug_scripts_section(cx: &CodegenCx<'_, '_>) -> bool {
     let omit_gdb_pretty_printer_section =
         find_attr!(cx.tcx.hir_krate_attrs(), AttributeKind::OmitGdbPrettyPrinterSection);
 
-    // We collect pretty printers transitively for all crates, so we make sure
-    // that the section is only emitted for leaf crates.
-    let embed_visualizers = cx.tcx.crate_types().iter().any(|&crate_type| match crate_type {
-        CrateType::Executable | CrateType::Cdylib | CrateType::Staticlib | CrateType::Sdylib => {
-            // These are crate types for which we will embed pretty printers since they
-            // are treated as leaf crates.
-            true
-        }
-        CrateType::ProcMacro => {
-            // We could embed pretty printers for proc macro crates too but it does not
-            // seem like a good default, since this is a rare use case and we don't
-            // want to slow down the common case.
-            false
-        }
-        CrateType::Rlib | CrateType::Dylib => {
-            // Don't embed pretty printers for these crate types; the compiler
-            // can see the `#[debug_visualizer]` attributes when using the
-            // library, and emitting `.debug_gdb_scripts` regardless would
-            // break `#![omit_gdb_pretty_printer_section]`.
-            false
-        }
-    });
+    // We could embed pretty printers for proc macro crates too but it does not
+    // seem like a good default, since this is a rare use case and we don't
+    // want to slow down the common case.
+    let proc_macro_only =
+        cx.tcx.crate_types().iter().all(|&crate_type| crate_type == CrateType::ProcMacro);
```

Comment:
I think removing this exception would make sense. Proc macros are generally compiled with debuginfo disabled nowadays, so performance isn't really a reason not to include the section for proc macros anymore.

Reply:
Good to know - removed :+1: